server/service/src/programs/program_event.rs

```diff
@@ -873,4 +881,148 @@ mod test {
             ]
         )
     }
+
+    fn check_integrity(mut events: Vec<ProgramEventRow>) {
```

Comment:
This method could be used in other tests to do a general integrity check but I only use it in the new test...